CHANGELOG.md

```diff
@@ -9,6 +9,8 @@ Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.1.0/)
 
 ### Enhancements
 ### Bug Fixes
+- Fix error codes ([#1083](https://github.com/opensearch-project/flow-framework/pull/1083))
```

Comment:
These aren't all error codes.  Should probably change the title to "Changed REST status codes for RBAC and provisioning"